### PR TITLE
Mepts | 1005 MQ Cat 15 Change Request

### PR DESCRIPTION
**Pull request by eliazmuchave**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java [hunk 1]
```diff
@@ -1978,6 +1981,13 @@ public CohortDefinition getMI15EComplete() {
    */
   public CohortDefinition getMI15I() {
 
+    CohortDefinition cd = getMI15I(18, 12);
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java:1984]: @eliazmuchave now it's 18 and 24 months

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java [hunk 2]
```diff
@@ -2053,7 +2070,7 @@ public CohortDefinition getCat15P1DenNum(boolean isDenominator, int level) {
 
     CohortDefinition a = getMI15A();
     CohortDefinition b1 = getMI15B1();
-    CohortDefinition b2 = getMI15B2();
+    CohortDefinition b2 = getMI15B2(21);
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java:2073]: @eliazmuchave it's 24 now

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java [hunk 3]
```diff
@@ -7353,6 +7566,143 @@ private CohortDefinition getAgeOnObsDatetime(Integer minAge, Integer maxAge) {
...
+        "Mq15I",
+        EptsReportUtils.map(
+            Mq15I, "startDate=${startDate},endDate=${revisionEndDate},location=${location}"));
+
+    cd.setCompositionString("Mq15MdsDen15 AND Mq15I");
+
+    return cd;
+  }
+
+  /** */
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java]: Can you please put the description from the spec you implementing here @eliazmuchave

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java [hunk 4]
```diff
@@ -7353,6 +7566,143 @@ private CohortDefinition getAgeOnObsDatetime(Integer minAge, Integer maxAge) {
...
+            + "                  GROUP  BY p.patient_id) AS breastfeeding "
+            + "              ON breastfeeding.patient_id = pregnant.patient_id "
+            + "WHERE  pregnant.pregnancy_date >= breastfeeding.breastfeed_date OR breastfeeding.breastfeed_date IS NULL "
+            + "GROUP  BY pregnant.patient_id";
+
+    StringSubstitutor stringSubstitutor = new StringSubstitutor(map);
+    sqlCohortDefinition.setQuery(stringSubstitutor.replace(query));
+    return sqlCohortDefinition;
+  }
+
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java:7656]: Same here, description @eliazmuchave

**Review by psmatsinhe — CHANGES_REQUESTED** [review]
Hi @eliazmuchave please have a look on the requests

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java (hunk 3 above)
Comment (eliazmuchave) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java]: I'll do it

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java (hunk 2 above)
Comment (eliazmuchave) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/IntensiveMonitoringCohortQueries.java:2073]: It's 24 months for new MQ indicator 15 but for MI it still 21 months

**Review by psmatsinhe — APPROVED** [review]
Looks good to me

**Pull request merged**